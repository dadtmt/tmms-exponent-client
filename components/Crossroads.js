import React, { PropTypes } from 'react'
import { View } from 'react-native'

import Crossroad from './Crossroad.js'

const Crossroads = ({ crossroads, makeChoice, resolveTest }) => <View>
  {
    crossroads.edges.map(({ node }) => <Crossroad
      key={node.id}
      {...node}
      makeChoice={makeChoice}
      resolveTest={resolveTest}
    />)
  }
</View>

Crossroads.propTypes = {
  crossroads: PropTypes.shape({
    edges: PropTypes.array.isRequired
  }).isRequired,
  makeChoice: PropTypes.func.isRequired,
  resolveTest: PropTypes.func.isRequired
}

export default Crossroads
