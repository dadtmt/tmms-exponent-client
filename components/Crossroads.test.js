import React from 'react'
import renderer from 'react-test-renderer'

import Crossroads from './Crossroads'

it('renders without crashing', () => {
  const text = {
    blocks: [
      {
        data: {},
        depth: 0,
        entityRanges: [],
        inlineStyleRanges: [],
        key: '4bkmd',
        text: 'some rich text',
        type: 'unstyled'
      }
    ],
    entityMap: {}
  }
  const props = {
    crossroads: { edges: [{
      node: {
        id: 'SOME_ID',
        choices: { edges: [] },
        testDices: { edges: [{
          node: {
            details: 'details',
            id: 'SOME_ID',
            made: false,
            master: false,
            modifier: 0,
            nbDices: 1,
            nbSides: 6,
            result: 6,
            text
          }
        }] },
        text
      }
    }] },
    makeChoice: jest.fn(),
    resolveTest: jest.fn()
  }
  expect(
    renderer.create(<Crossroads {...props} />).toJSON()
  ).toMatchSnapshot()
})
