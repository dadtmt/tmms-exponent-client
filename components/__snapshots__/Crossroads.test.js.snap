exports[`test renders without crashing 1`] = `
<View>
  <View>
    <View
      style={
        Object {
          "marginBottom": 20,
        }
      }>
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "fontSize": 15,
            }
          }>
          some rich text
        </Text>
      </View>
    </View>
    <View />
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#ccc",
            "marginBottom": 10,
            "padding": 5,
          }
        }>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "fontSize": 15,
                }
              }>
              some rich text
            </Text>
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Lancer les d√©s: 
            1D6+0
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
